spec/controllers/lesson_controller_spec.rb

```diff
@@ -0,0 +1,170 @@
+require 'rails_helper'
+require 'pry'
+
+RSpec.describe LessonsController, type: :controller do
+  describe 'routes' do
+    it { should route(:get, '/lessons').to(action: :index) }
+    it { should route(:get, '/lessons/574d5c29-fd2d-4494-ae28-12cdb0da1c07').to(action: :show, id: "574d5c29-fd2d-4494-ae28-12cdb0da1c07") }
+    it { should route(:post, '/lessons').to(action: :create) }
+    it { should route(:put, '/lessons/574d5c29-fd2d-4494-ae28-12cdb0da1c07').to(action: :update, id: "574d5c29-fd2d-4494-ae28-12cdb0da1c07") }
+    it { should route(:patch, '/lessons/574d5c29-fd2d-4494-ae28-12cdb0da1c07').to(action: :update, id: "574d5c29-fd2d-4494-ae28-12cdb0da1c07") }
+    it { should route(:delete, '/lessons/574d5c29-fd2d-4494-ae28-12cdb0da1c07').to(action: :destroy, id: "574d5c29-fd2d-4494-ae28-12cdb0da1c07") }
+  end
+
+  describe 'GET #index' do
+    subject do
+      get :index
+    end
+
+    before do
+      FactoryBot.create_list(:lesson, 2)
+      subject
+    end
+
+    it 'returns status 200' do
+      expect(response.code).to eq("200")
+    end
+
+    it 'lists 2 lessons' do
+      expect(json_response.count).to eq(2)
+    end
+  end
+
+  describe 'GET #show' do
+    let!(:lesson){ FactoryBot.create(:lesson) }
+    let(:id){ Lesson.last.id }
+
+    subject{ get :show, params: { id: id } }
+
+    context 'when lesson exist' do
+      it 'returns status 200' do
+        subject
+        expect(response.code).to eq("200")
+      end
+      it "shows the lesson" do
+        subject
+        expect(json_response[:id]).to eq(lesson.id)
+      end
+    end
+
+    context 'when lesson doesn\'t exist' do
+      let(:id){ 1 }
+      it "returns error 404" do
+        subject
+        expect(response.code).to eq("404")
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    let(:lesson_params){ FactoryBot.attributes_for(:lesson) }
+
+    subject{ post :create, params: { lesson: lesson_params } }
+
+    context 'with valid params' do
+      it 'returns status 201' do
+        subject
+        expect(response.code).to eq("201")
+      end
+
+      it 'creates new lesson' do
+        expect{ subject }.to change{ Lesson.count }.by(1)
+      end
+    end
+
+    context 'with invalid params' do
+      it 'returns status 403 without title' do
+        lesson_params[:title] = ''
+        subject
+        expect(response.code).to eq("403")
+      end
+      it 'doesn\'t create new lesson' do
+        lesson_params[:title] = ''
+        expect{ subject }.to change { Lesson.count }.by(0)
```

Comment:
Better written?
```ruby
expect{ subject }.to_not change { Lesson.count }
```
see: https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/change-matcher#expect-no-change

Reply:
Of course ! Done.